cmd/api/server_test.go

```diff
@@ -0,0 +1,200 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/dolittle/platform-api/cmd/api/mocks"
+	"github.com/dolittle/platform-api/pkg/k8s"
+	"github.com/dolittle/platform-api/pkg/platform/containerregistry"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+type response struct {
+	status int
+	field  string
+	value  interface{}
+}
+
+type request struct {
+	path   string
+	secret string
+}
+
+func expect(req request, res response) {
+	c := http.Client{}
+	httpReq, _ := http.NewRequest("GET", req.path, nil)
+	if req.secret == "" {
+		httpReq.Header.Del("x-shared-secret")
+	} else {
+		httpReq.Header.Set("x-shared-secret", req.secret)
+	}
+	httpReq.Header.Set("Tenant-ID", "123")
+	httpReq.Header.Set("User-ID", "666")
+
+	httpRes, _ := c.Do(httpReq)
+	Expect(httpRes).ToNot(BeNil())
+
+	if res.status != 0 {
+		Expect(httpRes.StatusCode).To(Equal(res.status))
+	} else {
+		Expect(httpRes.StatusCode).To(Equal(http.StatusOK))
+	}
+
+	if res.value != nil {
+		body, _ := ioutil.ReadAll(httpRes.Body)
+		var jsonData map[string]interface{}
+		json.Unmarshal(body, &jsonData)
+		fmt.Println(jsonData)
+		Expect(jsonData[res.field]).To(Equal(res.value))
+	}
+}
+
+var _ = Describe("Platform API", func() {
+	var gitRepo mocks.GitStorageRepoMock
+	var containerRegistryRepo *mocks.ContainerRegistryMock
+	var server *httptest.Server
+	var paths map[string]string
+	now := time.Now()
+
+	BeforeEach(func() {
+		viper.Set("tools.server.gitRepo.branch", "foo")
+		viper.Set("tools.server.gitRepo.directory", "/tmp/dolittle_operation")
+		viper.Set("tools.server.gitRepo.url", "git@github.com:dolittle-platform/Operations")
+		viper.Set("tools.server.gitRepo.sshKey", "does/not/exist")
+		viper.Set("tools.server.kubernetes.externalClusterHost", "external-host")
+		viper.Set("tools.server.secret", "johnc")
+
+		gitRepo = mocks.GitStorageRepoMock{}
+		containerRegistryRepo = &mocks.ContainerRegistryMock{}
+		containerRegistryRepo.StubAndReturnImages([]string{"helloworld"})
+		containerRegistryRepo.StubAndReturnTags(([]string{"latest", "v1"}))
+		containerRegistryRepo.StubAndReturnTags2([]containerregistry.ImageTag{{
+			Name:           "label1",
+			Digest:         "sha256:...",
+			CreatedTime:    now,
+			LastUpdateTime: now,
+			Signed:         false,
+		}})
+
+		logContext := logrus.StandardLogger()
```

Comment:
```suggestion
		logContext, _ = logrusTest.NewNullLogger()
```

You can also use the null logger from this package (the returned type from `NewNullLogger()` might be wrong but you get the idea):
```go
	logrusTest "github.com/sirupsen/logrus/hooks/test"
```